app/scripts/lib/transaction/institutional-snap/deferred-publication-hooks.ts

```diff
@@ -0,0 +1,186 @@
+    };
+  };
+  type: string;
+  fulfilled: boolean;
+  rejected: boolean;
+  lastUpdated: number;
+  transaction: {
+    custodianTransactionId: string;
+    transactionStatus: {
+      finished: boolean;
+      success: boolean;
+      displayText: string;
+      submitted: boolean;
+      reason: string;
+      signed: boolean;
+    };
+    from: string;
+    custodianPublishesTransaction: boolean;
+    maxFeePerGas: string;
+    maxPriorityFeePerGas: string;
+    gasLimit: string;
+    nonce: string;
+    to: string;
+    transactionHash: string;
+    type: string;
+  };
+  result: {
+    v: string;
+    r: string;
+    s: string;
+  };
+};
+
+export type InstitutionalSnapRequestSearchParameters = {
+  from: string;
+  to: string;
+  value: string;
+  data: string;
+  chainId: string;
+};
+
+type AllowedActions =
+  | HandleSnapRequest
+  | AccountsControllerGetAccountByAddressAction;
+
+export type DeferredPublicationHookMessenger = RestrictedMessenger<
+  'DeferredPublicationHookMessenger',
+  AllowedActions,
+  never,
+  AllowedActions['type'],
+  never
+>;
+
+type DeferrableTransactionAccount = {
+  options: {
+    custodian: {
+      deferPublication: boolean;
+    };
+  };
+};
+
+type DeferredPublicationControllerState = Record<string, never>;
+
+const metadata = {};
+
+export class DeferredPublicationController extends BaseController<
+  'DeferredPublicationController',
+  DeferredPublicationControllerState,
+  DeferredPublicationHookMessenger
+> {
+  constructor({ messenger }: { messenger: DeferredPublicationHookMessenger }) {
+    super({
+      messenger,
+      name: 'DeferredPublicationController',
+      state: {},
+      metadata,
+    });
+  }
+
+  private async handleSnapRequest(args: SnapRPCRequest) {
+    const response = await this.messagingSystem.call(
+      'SnapController:handleRequest',
+      args,
+    );
+    return response as InstitutionalSnapResponse;
+  }
+
+  private async shouldDeferPublication(transactionMeta: TransactionMeta) {
+    const account = (await this.messagingSystem.call(
+      'AccountsController:getAccountByAddress',
+      transactionMeta.txParams.from as string,
+    )) as unknown as DeferrableTransactionAccount;
+
+    return account?.options.custodian?.deferPublication;
+  }
+
+  async deferPublicationHook(
+    transactionController: TransactionController,
+    transactionMeta: TransactionMeta,
+  ): Promise<boolean> {
+    const shouldDefer = await this.shouldDeferPublication(transactionMeta);
+
+    if (shouldDefer) {
+      const searchParams: InstitutionalSnapRequestSearchParameters = {
+        from: transactionMeta.txParams.from as string,
+        to: transactionMeta.txParams.to as string,
+        value: transactionMeta.txParams.value as string,
+        data: transactionMeta.txParams.data as string,
+        chainId: transactionMeta.chainId as string,
+      };
+
+      const snapGetMutableTransactionParamsPayload: SnapRPCRequest = {
+        snapId,
+        origin: 'metamask',
```

Comment:
Minor, we have a constant for this called `ORIGIN_METAMASK` from `@metamask/controller-utils`.